### ISSUE TITLE
Discard edits message not showing when I make a change in a record then without clicking update button select another record

### ISSUE THREAD

**Issue by azizaparveen** [Esri/solutions-components#970]
### Describe the bug

Discard edits message not showing when I make a change in a record then without clicking update button select another record.
It is not showing the Discard edits message

### Expected Behavior

It should show the discard edits message since we are getting away from the Editing state without committing the edit.

### Reproduction Steps & Sample

Test app: https://solutions.mapsdevext.arcgis.com/apps/instant/manager/index.html?appid=fddd9b183ae14ffb9621b3fa51f852b8
1) Launch the test app
2) Select a record
3) Click Edit button
4) Make some changes 
5) Instead of clicking on Update, select another record (key step)

**Comment by azizaparveen** [comment]
cc @sarahmcdonald7

**Comment by sarahmcdonald7** [comment]
@azizaparveen Just a heads up, I don't think this is a regression. I was able to reproduce this on production. We should show that discard edits message if the app leaves an open edit form.

**Comment by azizaparveen** [comment]
@sarahmcdonald7 you are right, it is not a regression. John is working on it.

**Comment by azizaparveen** [comment]
Pushing this next release @sarahmcdonald7 @jmhauck @skrishn

**Comment by jmhauck** [comment]
@azizaparveen you will notice an issue when you are editing a feature and change layers and choose to continue editing. The layer selection in the layer picker is not showing the correct state. I will revisit this after the migration to calcite 3 is complete.

**Comment by jmhauck** [comment]
@azizaparveen your video is showing a new issue that we should fix in a patch. If you choose to continue editing the other layer should not be loading. Rather the layer the was initially loaded should still be loaded and the editor should remain active with any temp edits you have made. We have been using a method that was recommended by dev but I believe something in the editor has changed recently. I have followed up with dev about this but after some digging I see that we could potentially call another method that would trigger the expected behavior for this new issue.

**Comment by jmhauck** [comment]
In regards to 

> [@azizaparveen](https://github.com/azizaparveen) you will notice an issue when you are editing a feature and change layers and choose to continue editing. The layer selection in the layer picker is not showing the correct state. I will revisit this after the migration to calcite 3 is complete.

The problem is that the state of the layer picker shows both the layer that was initially selected and being edited as well as the other layer that was selected that caused the discard prompt to show are both shown as selected: 

![Image](https://github.com/user-attachments/assets/4a39484f-6ca7-41e9-80ed-64af7409a567)

**Comment by jmhauck** [comment]
@azizaparveen yes...would like to get confirmation from dev before making the commit but we should be good for patch A I believe.

**Comment by sarahmcdonald7** [comment]
> Sounds good [@jmhauck](https://github.com/jmhauck) , is it a small change, can we do it in Patch A?

@azizaparveen If this is going to be patched for A can you create an issue in the templates repo?

**Comment by azizaparveen** [comment]
> > Sounds good [@jmhauck](https://github.com/jmhauck) , is it a small change, can we do it in Patch A?
> 
> [@azizaparveen](https://github.com/azizaparveen) If this is going to be patched for A can you create an issue in the templates repo?

Yes, will do it in a min.

**Comment by azizaparveen** [comment]
Closing it, the state issue is a known issue.

![Image](https://github.com/user-attachments/assets/5a3ad94b-c731-4d50-bf12-90aacb7cb23b)
